lib/path.js

```diff
@@ -115,8 +138,8 @@ win32.resolve = function() {
 
     var result = splitDeviceRe.exec(path),
         device = result[1] || '',
-        isUnc = device && device.charAt(1) !== ':',
-        isAbsolute = win32.isAbsolute(path),
+        isUnc = device && device[1] !== ':',
+        isAbsolute = isUnc || result[2],
```

Comment:
We could have another function similar to [`win32SplitPath`](https://github.com/joyent/node/blob/v0.12/lib/path.js#L66) that returns the following properties:
+ `device`
+ `isUnc`
+ `isAbsolute`
+ `tail`

Also, is there any reason why [`win32SplitPath`](https://github.com/joyent/node/blob/v0.12/lib/path.js#L66) returns an array? I feel like the code would be more readable if it returned an object instead.